corehq/apps/userreports/indicators/specs.py

```diff
@@ -97,3 +97,18 @@ class ChoiceListIndicatorSpec(PropertyReferenceIndicatorSpecBase):
 
     def get_operator(self):
         return IN_MULTISELECT if self.select_style == 'multiple' else EQUAL
+
+
+class LedgerBalancesIndicatorSpec(IndicatorSpecBase):
+    type = TypeProperty('ledger_balances')
+    product_codes = ListProperty(required=True)
+    ledger_section = StringProperty()
+    case_id_expression = DictProperty()
+
+    def get_case_id_expression(self):
+        from corehq.apps.userreports.expressions.factory import ExpressionFactory
+        expression_spec = self.case_id_expression or {
+            "type": "property_name",
+            "property_name": "_id",
+        }
+        return ExpressionFactory.from_spec(expression_spec)
```

Comment:
Just tried out moving it to the `wrap` method - it looks like when editing a data source config, the indicators are wrapped during validation, but the wrapped results aren't saved, so moving this to `wrap` would require some reworking of that initial save.

Since having defaults is inconsistent with most of the UCR stuff, I just dropped it.